src/main/java/org/kairosdb/core/http/rest/MetricsResource.java

```diff
@@ -661,20 +664,28 @@ private Response executeNameQuery(NameType type) {
 		int time = m_ingestTime.getAndSet(0);
 		int count = m_ingestedDataPoints.getAndSet(0);
 
-		if (count == 0)
-			return Collections.emptyList();
-
 		DataPointSet dpsCount = new DataPointSet(INGEST_COUNT);
 		DataPointSet dpsTime = new DataPointSet(INGEST_TIME);
+		DataPointSet dpsRequestedCount = new DataPointSet(QUERY_DATAPOINTS_REQUESTED);
 
 		dpsCount.addTag("host", hostName);
 		dpsTime.addTag("host", hostName);
+		dpsRequestedCount.addTag("host", hostName);
+		dpsRequestedCount.addTag("artifact_version", m_artifactVersion);
+		dpsRequestedCount.addTag("deployment_id", m_deploymentId);
 
 		dpsCount.addDataPoint(m_longDataPointFactory.createDataPoint(now, count));
 		dpsTime.addDataPoint(m_longDataPointFactory.createDataPoint(now, time));
+		dpsRequestedCount.addDataPoint(m_longDataPointFactory.createDataPoint(now, m_datapointsCount));
 		List<DataPointSet> ret = new ArrayList<DataPointSet>();
-		ret.add(dpsCount);
-		ret.add(dpsTime);
+
+		if (count == 0){
```

Comment:
Anyways you are going to add the `dpsRequestedCount` irrespective of `count` So you can replace with:
```
ret.add(dpsRequestedCount);
	if (count != 0){
 			ret.add(dpsCount);
 			ret.add(dpsTime);
 	}
	return ret;
```